resources/sdk/clisdkclient/templates/restclient.mustache

```diff
@@ -0,0 +1,192 @@
+package restclient
+
+import (
+	"net/http"
+
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"gc/config"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/tidwall/pretty"
+)
+
+var (
+	Client HTTPClient
+)
+
+// HTTPClient interface
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type HttpStatusError struct {
+	Verb       string
+	Path       string
+	Body       string
+	StatusCode int
+}
+
+func (e *HttpStatusError) Error() string {
+	return fmt.Sprintf("HttpStatusCode error encountered while processing request to Genesys Cloud API. Verb %s, Path %s, Status Code %d, Body %v", e.Verb, e.Path, e.StatusCode, e.Body)
+}
+
+type RESTClient struct {
+	environment string
+	token       string
+}
+
+type oAuthToken struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   string `json:"expires_in"`
+	Error       string `json:"error"`
+}
+
+//Get executes an HTTP Get
+func (r *RESTClient) Get(uri string) (string, error) {
+	return r.callAPI(http.MethodGet, uri, "")
+}
+
+//Put executes an HTTP Put
+func (r *RESTClient) Put(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPut, uri, data)
+}
+
+//Post executes an HTTP Post
+func (r *RESTClient) Post(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPost, uri, data)
+}
+
+//Patch executes an HTTP Patch
+func (r *RESTClient) Patch(uri string, data string) (string, error) {
+	return r.callAPI(http.MethodPatch, uri, data)
+}
+
+//Delete executes an HTTP Delete
+func (r *RESTClient) Delete(uri string) (string, error) {
+	return r.callAPI(http.MethodDelete, uri, "")
+}
+
+func (r *RESTClient) callAPI(method string, uri string, data string) (string, error) {
+	var apiURI *url.URL
+	if !strings.HasPrefix(uri, "/") {
+		uri = fmt.Sprintf("/%v", uri)
+	}
+	if !strings.Contains(r.environment, "localhost") {
```

Comment:
@carnellj I added support for running against a locally hosted API here and made a small change to the Login function also.

Reply:
Cool